src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php

```diff
@@ -176,19 +179,28 @@ protected function getContainerBuilder()
             return $this->containerBuilder;
         }
 
-        if (!$this->getApplication()->getKernel()->isDebug()) {
+        $kernel = $this->getApplication()->getKernel();
+
+        if (!$kernel->isDebug()) {
             throw new \LogicException('Debug information about the container is only available in debug mode.');
         }
 
-        if (!is_file($cachedFile = $this->getContainer()->getParameter('debug.container.dump'))) {
-            throw new \LogicException('Debug information about the container could not be found. Please clear the cache and try again.');
+        if (is_file($cachedFile = $this->getContainer()->getParameter('debug.container.dump'))) {
```

Comment:
AFAIK, `debug.container.dump` was only added by the compiler pass you deprecated. So this looks suspicious to me